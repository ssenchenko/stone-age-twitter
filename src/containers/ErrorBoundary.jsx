import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import logger from '../services/logger';

const ErrorMessageStyled = styled.h1`
  color: #ef45ae;
  font-weight: bold;

  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

class ErrorBoundary extends Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  // eslint-disable-next-line no-unused-vars
  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, info) {
    logger.error(error, info);
  }

  render() {
    const { hasError } = this.state;
    if (hasError) {
      return (
        <ErrorMessageStyled>
          Something went wrong.
        </ErrorMessageStyled>
      );
    }
    const { children } = this.props;
    return children;
  }
}

export default ErrorBoundary;
