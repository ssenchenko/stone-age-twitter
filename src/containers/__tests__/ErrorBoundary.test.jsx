import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';

import ErrorBoundary from '../ErrorBoundary';

// no output needed
console.error = jest.fn(); // eslint-disable-line no-console

const Error = () => {
  throw new Error('Error thrown from problem child');
};

describe('ErrorBoundary', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <Error />
        </ErrorBoundary>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
