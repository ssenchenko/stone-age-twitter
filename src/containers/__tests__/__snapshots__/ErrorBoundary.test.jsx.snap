// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders correctly 1`] = `
.c0 {
  color: #ef45ae;
  font-weight: bold;
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

<h1
  className="c0"
>
  Something went wrong.
  <br />
  Error is not a constructor
</h1>
`;
